data_pipelines/rne/database/process_rne.py

```diff
@@ -0,0 +1,294 @@
+        CREATE TABLE IF NOT EXISTS dirigeants_pp
+        (
+            siren TEXT,
+            date_mise_a_jour TEXT,
+            date_de_naissance TEXT,
+            role TEXT,
+            nom TEXT,
+            nom_usage TEXT,
+            prenoms TEXT,
+            genre TEXT,
+            nationalite TEXT,
+            situation_matrimoniale TEXT,
+            pays TEXT,
+            code_pays TEXT,
+            code_postal TEXT,
+            commune TEXT,
+            code_insee_commune TEXT,
+            voie TEXT,
+            file_name TEXT
+        )
+    """
+    )
+    cursor.execute(
+        """
+        CREATE TABLE IF NOT EXISTS dirigeants_pm
+        (
+            siren TEXT,
+            date_mise_a_jour TEXT,
+            denomination TEXT,
+            siren_dirigeant TEXT,
+            role TEXT,
+            forme_juridique TEXT,
+            pays TEXT,
+            code_pays TEXT,
+            code_postal TEXT,
+            commune TEXT,
+            code_insee_commune TEXT,
+            voie TEXT,
+            file_name TEXT
+        )
+    """
+    )
+    create_index_db(cursor)
+
+
+def create_index_db(cursor):
+    index_statements = [
+        "CREATE INDEX IF NOT EXISTS idx_siren_pp ON dirigeants_pp (siren);",
+        "CREATE INDEX IF NOT EXISTS idx_siren_pm ON dirigeants_pm (siren);",
+        "CREATE INDEX IF NOT EXISTS file_pp ON dirigeants_pp (file_name);",
+        "CREATE INDEX IF NOT EXISTS file_pm ON dirigeants_pm (file_name);",
+        """CREATE INDEX IF NOT EXISTS idx_pp_siren_file_name
+        ON dirigeants_pp (siren, file_name);""",
+        """CREATE INDEX IF NOT EXISTS idx_pm_siren_file_name
+        ON dirigeants_pm (siren, file_name);""",
+    ]
+
+    for statement in index_statements:
+        cursor.execute(statement)
+
+
+def inject_records_into_db(file_path, db_path, file_type):
+    dirigeants_pp, dirigeants_pm = [], []
+
+    with open(file_path, "r") as file:
+        logging.info(f"Processing stock file: {file_path}")
+        try:
+            if file_type == "stock":
+                json_data = file.read()
+                data = json.loads(json_data)
+                for record in data:
+                    list_dirigeants_pp, list_dirigeants_pm = extract_dirigeants_data(
+                        record, file_type
+                    )
+                    dirigeants_pp = dirigeants_pp + list_dirigeants_pp
+                    dirigeants_pm = dirigeants_pm + list_dirigeants_pm
+            elif file_type == "flux":
+                for line in file:
+                    data = json.loads(line)
+                    for record in data:
+                        (
+                            list_dirigeants_pp,
+                            list_dirigeants_pm,
+                        ) = extract_dirigeants_data(record, file_type)
+                        dirigeants_pp = dirigeants_pp + list_dirigeants_pp
+                        dirigeants_pm = dirigeants_pm + list_dirigeants_pm
```

Comment:
very similar, you might want to factorize these two for loops.

Reply:
I tried in the previous commits (https://github.com/etalab/annuaire-entreprises-search-infra/pull/202/commits/5526c4c8993d8576028d8b6c1c36d670085104b6#) but I didn't succeed. As the double loop for `flux` files creates problems.
I might end up creating another helper to do this.

Reply:
Helper seems good